.github/workflows/test-updates.yml

```diff
@@ -0,0 +1,19 @@
+name: Test Updates
+on:
+  pull_request: {}
+  push: {}
+jobs:
+  updates:
+    name: Check if generated code is up-to-date
+    runs-on: [ubuntu-latest]
+    steps:
+    - name: Set up Go 1.17
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17
+    - name: Check-out code
+      uses: actions/checkout@v2
+    - name: Check updates
+      run: |
+        go install github.com/golang/mock/mockgen@v1.6.0
```

Comment:
maybe we need to make it clear somewhere in the doc that we are using mockgen v1.6.0, as the generated contents may vary with different versions.

Reply:
Sure. Added docs under `site/`

Reply:
I'm wondering if we can add a new make target to install mockgen and set it as a prerequisite for `make update`

Reply:
I will say let's put install mockgen into `update-mock-clients.sh`. Install mockgen doesn't seem to be cloud-provider-azure specific target. What's your idea?